### PR TITLE
N11 - unit tests

### PR DESCRIPTION
**Pull request by chiaryan**
Setup testing workflows for frontend and question-service

Unit tests for:
- frontend: Question Api
- question-service: `DecodeJSON` and `ReadQuestion`

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,8 +10,79 @@ on:
...
+        run: |
+          cd ./apps/frontend
+          pnpm i
+
+      - name: Run tests
+        run: |
+          cd ./apps/frontend
+          pnpm test
+
   test:
```
Comment (tituschewxj) [inline .github/workflows/test.yml]: Probably can rename this `test` to `docker-compose-tests`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -10,8 +10,79 @@ on:
...
+        run: |
+          cd ./apps/question-service
+          
+          echo "FIREBASE_CREDENTIAL_PATH=$QUESTION_FIREBASE_CREDENTIAL_PATH" >> .env
+          echo "JWT_SECRET=$JWT_SECRET" >> .env
+      
+      - name: Set up credentials
+        env:
+          QUESTION_FIREBASE_JSON: ${{ secrets.QUESTION_SERVICE_FIREBASE_CREDENTIAL }}
+          QUESTION_FIREBASE_CREDENTIAL_PATH: ${{ vars.QUESTION_SERVICE_FIREBASE_CREDENTIAL_PATH }}
```
Comment (tituschewxj) [inline .github/workflows/test.yml:36]: If we do not want to affect our staging or production databases, we could create another firestore just for testing purposes, and then create new github repository variables (e.g. `secrets.TEST_QUESTION_SERVICE_FIREBASE_CREDENTIAL`) for this purpose.

Then before every test is conducted, we just need to clear that database.

**Review by tituschewxj — APPROVED** [review]
LGTM! Thanks for implementing the tests. A suggestion would be to consider having a separate Firestore instance for testing.

File: .github/workflows/test.yml (hunk 1 above)
Comment (chiaryan) [reply .github/workflows/test.yml]: Ok, changed in latest commit

File: .github/workflows/test.yml (hunk 2 above)
Comment (chiaryan) [reply .github/workflows/test.yml:36]: - Workflow now starts and runs the go test against a local Firestore emulator, no need to setup another DB :D

https://firebase.google.com/docs/emulator-suite/install_and_configure

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -10,9 +10,81 @@ on:
...
+      - name: Install Go dependencies
+        run: |
+          cd ./apps/question-service
+          go mod tidy
+
+      - name: Install firebase tools
+        run: curl -sL firebase.tools | bash
+      
+      - name: Run Go tests with Firebase emulator
+        run: firebase emulators:exec --only firestore 'cd ./apps/question-service'
```
Comment (tituschewxj) [inline .github/workflows/test.yml]: This command should also include running the tests:

```suggestion
        run: firebase emulators:exec --only firestore 'cd ./apps/question-service && go test ./...'
```

Comment (chiaryan) [reply .github/workflows/test.yml]: oops. Ok fixed it, thanks for spotting this

**Pull request merged**